### ISSUE TITLE
[kws]: adding softmax layer will decrease accuracy for multihead attention rnn

### ISSUE THREAD

**Issue by lippman1125** [google-research/google-research#367]
@rybakov 

We use the training script listed [here](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_paper.md#att_mh_rnn).

https://github.com/google-research/google-research/blob/898f3471b88411e9e78155e4e5ca050c31c5c56b/kws_streaming/models/att_mh_rnn.py#L202-L203

The default param of `flags.return_softmax` is 0. 

https://github.com/google-research/google-research/blob/898f3471b88411e9e78155e4e5ca050c31c5c56b/kws_streaming/models/model_params.py#L70

If I use softmax, the accuracy will be decreased.

**Comment by rybakov** [comment]
Hello, thank you for evaluating kws lib! 

In this lib we use [SparseCategoricalCrossentropy(from_logits=True)](https://github.com/google-research/google-research/blob/master/kws_streaming/train/train.py#L82) loss, so return_softmax has to be equal zero. With this set up loss function computation may be more numerically stable as mentioned in [SparseCategoricalCrossentropy](https://www.tensorflow.org/api_docs/python/tf/keras/losses/SparseCategoricalCrossentropy)

return_softmax was introduced for special cases where loss function require probability distribution. For example if you change parameters of your loss function SparseCategoricalCrossentropy(from_logits=False) and set from_logits=False then you will need to enable return_softmax=1.

There is no need to modify return_softmax in this lib. For example if you plan to modify your loss function, only then you should consider to use return_softmax (depending on loss function parameters/config).
Please let me know if you have more questions.

**Comment by rybakov** [comment]
Thanks for reporting your numbers! It looks like I added a bug in my recent code refactor, that is why your accuracy is a little off. I am validating it now and will follow up on Monday.

**Comment by rybakov** [comment]
lippman1125@ I reverted a code refactor (which was compatible with eager mode only) please update your code with the latest [version](https://github.com/google-research/google-research/tree/master/kws_streaming). I trained att_mh_rnn model several time on data set v2 and get 98%
